Automattic/jetpack
# AI Assistant: Do not mark words starting with special characters as spelling mistakes

Fixes #39040

## Proposed changes:

* Exclude words starting with `@`, `#`, `$` and `+` from spelling mistake detection, including hyphenated words

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Go to the block editor
* Add some words detected as possible typos
* Add the same words, but starting with the special characters above, including hyphenated words like `#wabi-sabi`
* Check that those words are not highlighted

| Before | After |
| - | - |
| ![2024-08-22_18-45-06](https://github.com/user-attachments/assets/56eea505-e201-4177-8182-810d9cf73cb7) | ![2024-08-22_18-44-45](https://github.com/user-attachments/assets/ae7820a7-ed83-4ca1-b282-39dec7b2ea50) |
